src/views/ApartmentApp/components/StudentApplication/index.js

```diff
@@ -342,8 +342,8 @@ const StudentApplication = ({ userProfile, authentication }) => {
       );
       if (result) {
         try {
-          const result = loadApplication();
-          if (!result) {
+          const loadingResult = loadApplication();
```

Comment:
Shouldn't this be using an `await` as well?

Reply:
Yes, since `loadApplication()` is `async`, this should absolutely be using await, otherwise `loadingResult` will always evaluate to `true` since it's a `Promise`.

Reply:
Coming soon-ish in another branch, it is not as urgent as the larger bug fixes